PIT context relocation work on main repo

cbuescher opened the pull request:
Work to support PIT context relocation when nodes gracefully shut down and shards are relocated.

cbuescher committed:
PIT context relocation work on main repo

elasticsearchmachine commented:
Pinging @elastic/es-search-foundations (Team:Search Foundations)

cbuescher committed:
Update docs/changelog/137675.yaml

elasticsearchmachine commented:
Hi @cbuescher, I've created a changelog YAML for you.

cbuescher committed:
Add some logging and utility getter for tests

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Only retry searchable shards

fcofdez commented on x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/RetrySearchIntegTests.java:
```diff
@@ -43,6 +44,10 @@
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertNoFailuresAndResponse;
 import static org.hamcrest.Matchers.equalTo;
 
+@TestIssueLogging(
```
is this change related?

fcofdez reviewed (approved):
LGTM

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Add flag to mark ReaderContexts as expired immediately

This adds a kind of "dirty" flag to a ReaderContext that can be used to
make it immediately elgible for cleanup when the reaper process runs. We
cannot really achieve this using the current keepalive setting since keepalive
will be restarted every time a context is used. In some PIT relocation
scenarios we need to make a context "unusable" for new searches without
immediately removing it for running searches (i.e. when we remove a
context after a search has started, some phases like DFSQueryPhase don't
retry on missing contexts, so those searches would fail if we
immediately close the context).

cbuescher committed:
Acquire SearchSupplier through infrastructure in Engine

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
iter

Commit:
[CI] Auto commit changes from spotless

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Simplify changes in Engine a bit

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Merge branch 'main' into HEAD

cbuescher committed:
some cleanups

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Move changes in BlobFileRanges

cbuescher committed:
Add dynamic setting to enable pit relocation

cbuescher committed:
Register Setting

cbuescher committed:
Merge commit '88c87b0' into pit-relocation

cbuescher committed:
Merge branch 'main' into pit-relocation

cbuescher committed:
Merge commit 'e5652727f03a3cc08421b142d3a5f43da7cdf6d8' into HEAD

cbuescher committed:
Merge commit 'c4ecf776fb6131fa30eee85276d6450a16bdb151' into HEAD

cbuescher committed:
Merge commit '553eac1' into HEAD

cbuescher committed:
Merge commit '96fad57' into pit-relocation

cbuescher committed:
Adding SearchService to snapshot test helper
